pkg/project/auth/cache.go

```diff
@@ -67,6 +67,35 @@ func subjectRecordKeyFn(obj interface{}) (string, error) {
 	return subjectRecord.subject, nil
 }
 
+type State interface {
+	State() string
+}
+
+type reflectorState struct {
+	reflectors []*cache.Reflector
+}
+
+func (rs *reflectorState) State() string {
+	state := ""
+	for i := range rs.reflectors {
```

Comment:
Seems like collisions are technically possible if these were running against different etcds.  You could add a delimiter to eliminate the possibility.